src/dmd/traits.d

```diff
@@ -1080,8 +1080,34 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
                 auto fd = s.isFuncDeclaration();
                 if (!fd)
                 {
-                    if(includeTemplates)
-                        exps.push(new DsymbolExp(Loc.initial, s, false));
+                    if (includeTemplates)
+                    {
+                        if (auto td = s.isTemplateDeclaration())
+                        {
+                            // if td is part of an overload set we must take a copy
+                            // which shares the same `instances` cache but without
+                            // `overroot` and `overnext` set to avoid overload
+                            // behaviour in the result.
+                            if(td.overnext !is null)
```

Comment:
there's a space missing between the keyword and the left parens.

Reply:
I'm not doing well at this. Fixed now thanks.